service/history/timerQueueProcessor.go

```diff
@@ -238,16 +151,12 @@ func (t *timerQueueProcessorImpl) NotifyNewTimer(timerTasks []persistence.Task)
 			t.metricsClient.IncCounter(metrics.TimerTaskDeleteHistoryEvent, metrics.NewTimerCounter)
 		}
 	}
-	t.lock.Unlock()
 
-	if updatedMinTimer {
-		select {
-		case t.newTimerCh <- struct{}{}:
-			// Notified about new timer.
-
-		default:
-			// Channel "full" -> drop and move on, since we are using it as an event.
-		}
+	select {
+	case t.newTimerCh <- newEarlistTime:
```

Comment:
What happens if the earliest time is dropped due to non blocking write?

Reply:
changed back to use lock + variable for tracking min pending time